client/tink_keyset_helper.go

```diff
@@ -42,3 +47,37 @@ func nameOfSupportedTinkKeyTemplates() string {
 	sort.Strings(supportedTemplates)
 	return strings.Join(supportedTemplates, "\n")
 }
+
+// checkTemplateNameAndKnoxIDForTinkKeyset checks whether knox identifier start with "tink:<tink_primitive_short_name>:".
+func checkTemplateNameAndKnoxIDForTinkKeyset(templateName string, knoxIentifier string) error {
+	templateInfo, ok := tinkKeyTemplates[templateName]
+	if !ok {
+		return errors.New("not supported Tink key template. See 'knox key-templates'")
+	} else if !strings.HasPrefix(knoxIentifier, templateInfo.knoxIDPrefix) {
+		errInfo := fmt.Sprintf("<key_identifier> must have prefix '%s'", templateInfo.knoxIDPrefix)
+		return errors.New(errInfo)
+	}
+	return nil
+}
+
+// createNewTinkKeyset creates a new tink keyset contains a single fresh key from the given tink key templateFunc.
+func createNewTinkKeyset(templateFunc func() *tinkpb.KeyTemplate) []byte {
+	// Creates a keyset handle that contains a single fresh key
+	keysetHandle, err := keyset.NewHandle(templateFunc())
+	if keysetHandle == nil || err != nil {
+		fatalf("cannot get tink keyset handle: %v", err)
+	}
+	return convertTinkKeysetHandleToBytes(keysetHandle)
+}
+
+// convertTinkKeysetHandleToBytes extracts keyset from tink keyset handle and converts it to bytes
+func convertTinkKeysetHandleToBytes(keysetHandle *keyset.Handle) []byte {
+	bytesBuffer := new(bytes.Buffer)
+	writer := keyset.NewBinaryWriter(bytesBuffer)
+	// To write cleartext keyset handle, must use package "insecurecleartextkeyset"
+	err := insecurecleartextkeyset.Write(keysetHandle, writer)
+	if err != nil {
+		fatalf("cannot write tink keyset: %v", err)
```

Comment:
Same thing here. It'd be cleaner to return the error and trigger the panic at the top level.

Reply:
Thank you so much! I have revised my code according to your comment.